### ISSUE TITLE
Repository Intention

### ISSUE THREAD

**Issue by enckse** [redhat-plumbers/dracut-fedora#16]
What is the true relation between this (and the other redhat-plumbers) repositories and the actual upstream at this point? Fedora generally tracks spec files in [pagure](https://src.fedoraproject.org/rpms/dracut) and yet here there is a spec file and changes to the workflow actions here though this will not be part of a normal Fedora build/release at this time (isn't that just fully duplicate work at this point)?

Additionally this creates a lot of confusion about upstream/downstream and packaging. Is Fedora going to cut releases from this repository and therefore this is a full fork? Is the relationship then that [upstream](https://github.com/dracutdevs/dracut) will be merged in at points? Does Fedora (and RedHat) no longer work items upstream to support other distributions?

Is this repository actually used for anything at all or just a public testbed for ideas?

There is a lot of confusing information to be found here:
- https://src.fedoraproject.org/rpms/dracut/pull-request/54
- https://github.com/dracutdevs/dracut/pull/2103#issuecomment-1357532979

**Comment by pvalena** [comment]
Sorry, I didn't get around to writing a README yet. This is a source-git, as you've correctly identified [upstream PR](https://src.fedoraproject.org/rpms/dracut/pull-request/54), it was generated from this state (without any other manual work). So consider this on par with dist-git (inheritance should go both ways), as it should never diverge.

The plan is to have the changes fully automated, so when PR gets merged here (i.e. new commits land), a new PR with changes gets opened to pagure.

I've elaborated on a relation with upstream in the downstream PR, hopefully it's clear enough. There's no plan to diverge from upstream at all. Source-git actually helps getting feedback to upstream better (please see packit docs on source-git[1]).

**[1] https://packit.dev/source-git**

**Comment by AdamWill** [comment]
I use provenpackager privileges occasionally to push emergency fixes direct to dracut. I'm not a dracut maintainer in any regards, so it's not necessary to CC me on everything. I only care when stuff breaks.

**Comment by pvalena** [comment]
> This is some future plan, does not explain how to contribute to Fedora.

Yes, the plan is to have it in sync automatically - I currently do it manually (although I have some automation around it). Ideally, the contribution would be better here- where the test suite runs; at least before I (or some other volunteer) get around to set it up for Fedora CI.

> It seems some people directly push to https://src.fedoraproject.org/rpms/dracut/commits/rawhide, that are missing from this repository but yet you are asking dracut contributors to upload PRs here.

Yes, creating PR here is the preferred way, not the only way. I have not synced the changes back yet from the dist-git. (It's not overwritten, it's a two-way sync; although I'm working with packit team on some enhancements.)

> It seems PRs on https://src.fedoraproject.org/rpms/dracut/pull-requests are not landing - not even from RedHat employees :-)

Yeah, I have no time to review it; and don't want the packit automation, probably. I've left it open to investigate.

> Data tells a different story. Fedora Rawhide has taken every second release, which means feedback is delayed on a given upstream PR 7 month in some cases - see https://bugzilla.redhat.com/show_bug.cgi?id=2379116

Yeah, I was swamped elsewhere. It wouldn't be any different on just a dist-git. Should be definitely better in long-term. I'm more inclined to review and test things before I push it to Fedora - not just push and see what breaks.
But you're right, I was very delayed with 106/107 - that was not intentional.

**Comment by pvalena** [comment]
_I wonder, is initrd a space people really need to get updates as fast as possible? Are there features people long to get earlier, or are some major fixes introduced? I was always on the side of "my system boots, that's great" side of things with dracut. Am I wrong?_

**Comment by pvalena** [comment]
> > You're free to create PRs in either dist-git and source-git.
> 
> Despite you keep repeating it, this is not true as we can not update dracut to a newer version with a simple PR. Your RedHat colleges tried just to find out that those PRs are never reviewed - https://src.fedoraproject.org/rpms/dracut/pull-request/72 - 5 month ago, not even a simple feedback on them.

That's just an automated bump using packit; which is done automatically, and fails to build because it doesn't handle patches at all. I'm in regular contact with packit team.

> On other distribution - e.g. openSUSE and Debian, there are several maintainers even downstream for something as critical as dracut reviewing each other PRs. You seem to be arguing that this is not necessary for Fedora.

Am I?
